src/main/scala/viper/silver/plugin/standard/inline/InlinePredicatePlugin.scala

```diff
@@ -27,21 +27,25 @@ class InlinePredicatePlugin extends SilverPlugin with ParserPluginTemplate
 
   override def beforeVerify(input: Program): Program = {
     val rewrittenMethods = input.methods.map { method =>
-      val inlinePredIds = input.extensions.collect({case InlinePredicate(p) => p.name}).toSet
-      val recursivePreds = checkRecursive(inlinePredIds, input) ++ checkMutualRecursive(inlinePredIds, input)
-      // TODO: Do we also need to inline in inhale/exhale/assert/assume and package/apply statements?
-      val (prePredIds, postPredIds) = getPrePostPredIds(method, input, inlinePredIds)
-      val prePredIdsNoRecur = prePredIds.diff(recursivePreds.map(_.name))
-      val postPredIdsNoRecur = postPredIds.diff(recursivePreds.map(_.name))
-      val inlinedPredMethod = inlinePredicates(method, input, prePredIdsNoRecur, postPredIdsNoRecur)
-      rewriteMethod(inlinedPredMethod, input, prePredIdsNoRecur, postPredIdsNoRecur)
+      val allPredIds = input.predicates.filter(_.body.isDefined).map(_.name).toSet
```

Comment:
Just for the future, a sequence of `.filter` and `.map` can usually be represented by a `.collect`
```Scala
val allPredIds = input.predicates.collect {
    case pred if pred.body.isDefined => pred.name
}.toSet
```

Reply:
ooh neat, I'm gonna try that